controllers/storagecluster/external_resources.go

```diff
@@ -272,6 +272,9 @@ func (obj *ocsExternalResources) ensureCreated(r *StorageClusterReconciler, inst
 		r.Log.Error(err, "Could not create ExternalStorageClusterResource.", "StorageCluster", klog.KRef(instance.Namespace, instance.Name))
 		return reconcile.Result{}, err
 	}
+
+	// at this point the checksums are different, so update it
+	instance.Status.ExternalSecretHash = extSecretChecksum
```

Comment:
Instead of changing the function signature to return the hash we should just call r.externalSecretDataChecksum(instance) here to get the hash.